test/e2e/extended_suite_test.go

```diff
@@ -16,6 +17,16 @@ var (
 	kubeconfig = flag.String("kubeconfig", "../../_data/_out/admin.kubeconfig", "Location of the kubeconfig")
 )
 
+var _ = BeforeEach(func() {
```

Comment:
Needs to be moved back to were it was (inside the end user e2e tests). In other words, there should be no diff in your PR related to this.

Reply:
The problem with doing this for every suite is that in the azure cluster reader tests we don't need to create namespaces (at least so far) so the extra creation and deletion would incur considerable slowdown to our test times.